pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -976,21 +953,28 @@ func setBridgeIPAddr(
 			status.BridgeName)
 		return errors.New(errStr)
 	}
-	if status.Gateway != nil {
-		ipAddr = status.Gateway.String()
-		status.IPAssignments[bridgeMac.String()] = status.Gateway
-	}
-	log.Functionf("BridgeMac: %s, ipAddr: %s\n",
+	log.Functionf("BridgeMac: %s, ipAddr: %s",
 		bridgeMac.String(), ipAddr)
-	status.BridgeIPAddr = ipAddr
-	publishNetworkInstanceStatus(ctx, status)
-	log.Functionf("Published NetworkStatus. BridgeIpAddr: %s\n",
-		status.BridgeIPAddr)
 
-	if status.BridgeIPAddr == "" {
-		log.Functionf("Does not yet have a bridge IP address for %s\n",
-			status.Key())
-		return nil
+	// outside of subnet, flag it
+	if !status.Subnet.Contains(status.Gateway) {
+		errStr := fmt.Sprintf("Bridge IP(%s) is not in Subnet", ipAddr)
+		log.Errorf("setBridgeIPAddr(%s): fail: %s",
+			status.UUID.String(), errStr)
+		return errors.New(errStr)
+	}
+	// if inside of DhcpRange, flag it
+	ret, err := status.DhcpRange.Contains(status.Gateway)
+	if err != nil {
+		log.Errorf("setBridgeIPAddr(%s): fail: %s\n",
+			status.UUID.String(), err)
+		return err
+	}
+	if ret {
+		errStr := fmt.Sprintf("Bridge IP(%s) is in Dhcp Range(%s,%s)",
```

Comment:
Actually, the user error is that the Gateway is in the DHCP range (the user doesn't set anything called a bridge IP).